src/sshagent/AgentSettingsWidget.cpp

```diff
@@ -0,0 +1,36 @@
+/*
+ *  Copyright (C) 2017 Toni Spets <toni.spets@iki.fi>
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 or (at your option)
+ *  version 3 of the License.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "AgentSettingsWidget.h"
+#include "core/Config.h"
+
+AgentSettingsWidget::AgentSettingsWidget(QWidget* parent)
+    : QWidget(parent)
+    , m_ui(new Ui::AgentSettingsWidget())
+{
+    m_ui->setupUi(this);
+}
+
+void AgentSettingsWidget::loadSettings()
+{
+    m_ui->enableSSHAgentCheckBox->setChecked(config()->get("SSHAgent", false).toBool());
+}
+
+void AgentSettingsWidget::saveSettings()
+{
+    config()->set("SSHAgent", m_ui->enableSSHAgentCheckBox->isChecked());
```

Comment:
At this point you should emit a signal that the SSH Agent settings have changed which would alert the system to either start or stop the ssh agent. We should __never__ have to restart the application to enact a setting.

Reply:
This will require some refactoring as yeah, it needs to send the keys out to an agent if databases are unlocked when you enable it and emulate a lock if you disable while keys have been sent that are supposed to be removed on lock.

Reply:
We can leave that to a future enhancement after this merge. I'll create an issue to track.